irispy/iris_tools.py

```diff
@@ -672,15 +874,37 @@ def calculate_photons_per_sec_to_radiance_factor(
         for input wavelengths.
 
     """
+    # Get best fit by the method of least squares
+    iris_fit = fit_iris_xput(time_obs, time_cal_coeffs, cal_coeffs)  # Need to work on this...
     # Get effective area and interpolate to observed wavelength grid.
-    eff_area_interp = _get_interpolated_effective_area(detector_type, wavelength)
+    eff_area_interp = _get_interpolated_effective_area(iris_fit, detector_type, wavelength)
     # Return radiometric conversed data assuming input data is in units of photons/s.
-    return constants.h * constants.c / wavelength / u.photon / \
-           spectral_dispersion_per_pixel / eff_area_interp / solid_angle
+    return constants.h * constants.c / wavelength / u.photon / spectral_dispersion_per_pixel / eff_area_interp / solid_angle
 
-def _get_interpolated_effective_area(detector_type, obs_wavelength):
-    # Get effective area
-    ########### This needs to be generalized to the time of OBS once that functionality is written #########
+
+def _get_interpolated_effective_area(iris_fit, detector_type, obs_wavelength):
```

Comment:
The API of this version of this function should not include ```iris_fit```, but simply the time at which the effective area is required:
```python
def _get_interpolated_effective_area(detector_type, obs_wavelength, **kwargs):
```
where the ```**kwargs``` get fed to ```get_iris_reponse```.

Since this function is used in a few places, in the future we could look for a way to make it more efficient by not calling ```get_iris_response``` but storing the response in the object somewhere.  For now though, go with the above API.